containerd-shim-v2/service.go

```diff
@@ -403,7 +403,30 @@ func (s *service) Update(ctx context.Context, r *taskAPI.UpdateTaskRequest) (*pt
 
 // Wait for a process to exit
 func (s *service) Wait(ctx context.Context, r *taskAPI.WaitRequest) (*taskAPI.WaitResponse, error) {
-	return nil, errdefs.ErrNotImplemented
+	var ret uint32
+
+	s.Lock()
+	c, err := s.getContainer(r.ID)
+	s.Unlock()
+
+	if err != nil {
+		return nil, err
+	}
+
+	//wait for container
+	if r.ExecID == "" {
+		ret = <-c.exitCh
```

Comment:
I think here it would be very nice to add `s.sandbox.StopContainer()`. We want to make sure the container is completely stopped inside the VM. The container process might terminate because it properly returned, and we want to ensure there's nothing left inside the VM.

Reply:
Ah yes that's cool! Compared to the madness that we have to handle with our OCI CLI implementation, this is a real improvement here.

Reply:
Yes, if it's needed in the future  I'll add it.

Reply:
I had renamed it to netns and add  a line comment.